From 10587d1b9d037a0ccef46de91a56446d3791b21c Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 20 Jan 2021 23:15:30 +0200
Subject: [PATCH] [SW-30451] i2c indirect: added device id under info opcode

bmc controller can get the pci device id field as part of the device
info indirect opcode in preboot phase

Change-Id: I710f20148bdcb020be1b81b82e46cb78b67edf75
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/i2cslave_indirect.c b/preboot/i2cslave_indirect.c
index b624b7d..d3ffb8e 100644
--- a/preboot/i2cslave_indirect.c
+++ b/preboot/i2cslave_indirect.c
@@ -16,6 +16,7 @@
 #include "fit.h"
 
 #define SUPPORTED_OPCODES_MASK	( \
+				BIT(HL_I2CI_REQ_OP_DEVICE_INFO) | \
 				BIT(HL_I2CI_REQ_OP_DEVICE_STATUS) | \
 				BIT(HL_I2CI_REQ_OP_CONFIG_ACCESS) | \
 				BIT(HL_I2CI_REQ_OP_FW_LOADER) | \
@@ -53,6 +54,10 @@
 static struct hl_i2ci_device_status op_status = {
 	.device_status = HL_I2CI_REQ_DEV_STS_PREBOOT,
 	.valid_opcode_map = SUPPORTED_OPCODES_MASK,
+};
+
+static struct hl_i2ci_device_info op_info = {
+	.device_id = 0,
 };
 
 static struct hl_i2ci_config_access op_config;
@@ -241,6 +246,13 @@
 static inline void i2csi_req_process_read_opcode(void)
 {
 	switch (i2csi_req.wd.hdr.opcode) {
+	case HL_I2CI_REQ_OP_DEVICE_INFO:
+		if (!op_info.device_id)
+			op_info.device_id = (u16)(device_id_vendor_id >> 16);
+
+		i2csi_req.rd.data = (u8 *)&op_info;
+		i2csi_req.rd.length = sizeof(op_info);
+		break;
 	case HL_I2CI_REQ_OP_DEVICE_STATUS:
 		i2csi_req.rd.data = (u8 *)&op_status;
 		i2csi_req.rd.length = sizeof(op_status);
